File: 02 Ad-Hoc/3424 Achando os Monótonos Não-Triviais Maximais/3424.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N, tt = 0;
        boolean as_sqc = false;
        String sqc;
        N = scanner.nextInt();
        sqc = scanner.next();
        for (int i = 0; i < N; i++) {
            if (sqc.charAt(i) == 'a' && !as_sqc) {
                if (sqc.charAt(i + 1) == 'a')
                    tt++;
                as_sqc = true;
            } else if (sqc.charAt(i) == 'a' && as_sqc) {
                tt++;
            } else
                as_sqc = false;
        }
        System.out.println(tt);
        scanner.close();
    }
}

// https://www.beecrowd.com.br/judge/pt/problems/view/3424
// Ranking 00001º, Time 0.139, Submiss Date 29/05/2023 04:12:28 (Java 14)

File: 02 Ad-Hoc/3424 Achando os Monótonos Não-Triviais Maximais/3424.c
# include <stdio.h>

int main(){
  int N, tt = 0;
  int as_sqc = 0;
  scanf("%d", &N);
  char sqc[N];
  scanf(" %s", sqc);
  for (int i = 0; i < N; i++){
    if (sqc[i] == 'a' && as_sqc == 0){
      if (sqc[i+1] == 'a') tt++;
      as_sqc = 1;
    }
    else if (sqc[i] == 'a' && as_sqc == 1){
      tt++;
    }
    else as_sqc = 0;
  
  }
  
  printf("%d\n", tt);
}

// https://www.beecrowd.com.br/judge/pt/problems/view/3424
// Ranking 00001º, Time 0.000, Submiss Date 29/05/2023 07:03:22 (C99)